spec/classes/facts_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+
+describe 'structured_facts' do
+  family = 'RedHat'
+
+  context 'should compile using symbols' do
+    let(:facts) {{
+      :os => {
+        :family => family
+      }
+    }}
+
+    it { should contain_class('structured_facts') }
+    it { should compile.with_all_deps }
+  end
+
+  context 'should compile using strings' do
+    let(:facts) {{
+      'os' => {
+        'family' => 'RedHat'
+      }
+    }}
+
+    it { should contain_class('structured_facts') }
+    it { should compile.with_all_deps }
```

Comment:
Please add a test for the actual notifies here, so that the passed value in the notify title is actually checked for being correct.

something like

```
it { should contain_notify("RedHat") }
```

Also, I suspect that the notifies you create should cause a collision, because there should be two Notify[RedHat] resources?